app/core/memberships/endpoints_memberships.py

```diff
@@ -0,0 +1,499 @@
+    """
+
+    memberships = await cruds_memberships.get_association_memberships(db)
+    return memberships
+
+
+@router.get(
+    "/memberships/{association_membership_id}/members",
+    response_model=list[schemas_memberships.UserMembershipComplete],
+    status_code=200,
+)
+async def read_association_membership(
+    association_membership_id: uuid.UUID,
+    minimalStartDate: date = Query(None),
+    maximalStartDate: date = Query(None),
+    minimalEndDate: date = Query(None),
+    maximalEndDate: date = Query(None),
+    db: AsyncSession = Depends(get_db),
+    user: models_users.CoreUser = Depends(is_user_an_ecl_member),
+):
+    """
+    Return membership with the given ID.
+
+    **This endpoint is only usable by ECL members**
+    """
+
+    db_association_membership = (
+        await cruds_memberships.get_association_membership_by_id(
+            db=db,
+            membership_id=association_membership_id,
+        )
+    )
+    if db_association_membership is None:
+        raise HTTPException(status_code=404, detail="Association Membership not found")
+
+    db_user_memberships = (
+        await cruds_memberships.get_user_memberships_by_association_membership_id(
+            db=db,
+            association_membership_id=association_membership_id,
+            minimal_start_date=minimalStartDate,
+            maximal_start_date=maximalStartDate,
+            minimal_end_date=minimalEndDate,
+            maximal_end_date=maximalEndDate,
+        )
+    )
+
+    return db_user_memberships
+
+
+@router.post(
+    "/memberships/",
+    response_model=schemas_memberships.MembershipSimple,
+    status_code=201,
+)
+async def create_association_membership(
+    membership: schemas_memberships.MembershipBase,
+    db: AsyncSession = Depends(get_db),
+    user: models_users.CoreUser = Depends(is_user_in(GroupType.admin)),
+):
+    """
+    Create a new membership.
+
+    **This endpoint is only usable by administrators**
+    """
+    if await cruds_memberships.get_association_membership_by_name(
+        name=membership.name,
+        db=db,
+    ):
+        raise HTTPException(
+            status_code=400,
+            detail=f"A membership with the name {membership.name} already exists",
+        )
+
+    group = await cruds_groups.get_group_by_id(db, membership.group_id)
+    if group is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Group not found",
+        )
+
+    db_association_membership = schemas_memberships.MembershipSimple(
+        name=membership.name,
+        group_id=membership.group_id,
+        id=uuid.uuid4(),
+    )
+
+    cruds_memberships.create_association_membership(
+        db=db,
+        membership=db_association_membership,
+    )
+    try:
+        await db.commit()
+    except Exception:
+        raise HTTPException(
+            status_code=500,
+            detail="Failed to create membership",
+        )
```

Comment:
Why don't you let FastAPI catch and log the server error?
```suggestion
    await db.commit()
```